tests/test_file_processing.py

```diff
@@ -264,104 +263,116 @@ def test_row_required_fields_present(config):
         "",
         "",
     )
 
     # shorter plate barcode
-    assert centre_file.extract_plate_barcode_and_coordinate(
-        {"RNA ID": "DN1234567_H01"}, 0, barcode_field, barcode_regex
-    ) == (
+    short = {"RNA ID": "DN1234567_H01"}
+    assert centre_file.extract_plate_barcode_and_coordinate(short, 0, barcode_field, barcode_regex) == (
         "DN1234567",
         "H01",
     )
 
+    # from RT
+    rt = {"RNA ID": "`AP-abc-00020028_C11"}
+    assert centre_file.extract_plate_barcode_and_coordinate(rt, 0, barcode_field, barcode_regex) == (
+        "AP-abc-00020028",
+        "C11",
+    )
+
+    # white space and garbage characters
+    garbage = {"RNA ID": "   `£$%^&AP-abc-12345678_H01(*&^%$ ` `"}
+    assert centre_file.extract_plate_barcode_and_coordinate(garbage, 0, barcode_field, barcode_regex) == (
+        "AP-abc-12345678",
+        "H01",
+    )
+
 
 # tests for parsing and formatting the csv file rows
-def test_parse_and_format_file_rows(config):
-    timestamp = "some timestamp"
+def test_parse_and_format_file_rows(config, freezer):
+    now = datetime.now()
     test_uuid = uuid.uuid4()
     centre_file = centre_file_with_mocked_filtered_postitive_identifier(config, "some file")
-    with patch.object(centre_file, "get_now_timestamp", return_value=timestamp):
-        with patch("crawler.file_processing.uuid.uuid4", return_value=test_uuid):
-            extra_fields_added = [
-                {
-                    "Root Sample ID": "1",
-                    "RNA ID": "RNA_0043_H09",
-                    "plate_barcode": "RNA_0043",
-                    "source": "Alderley",
-                    "coordinate": "H09",
-                    "line_number": 2,
-                    "Result": "Positive",
-                    "file_name": "some file",
-                    "file_name_date": None,
-                    "created_at": timestamp,
-                    "updated_at": timestamp,
-                    "Lab ID": None,
-                    "filtered_positive": True,
-                    "filtered_positive_version": "v2.3",
-                    "filtered_positive_timestamp": timestamp,
-                    "lh_sample_uuid": str(test_uuid),
-                }
-            ]
+    with patch("crawler.file_processing.uuid.uuid4", return_value=test_uuid):
+        extra_fields_added = [
+            {
+                "Root Sample ID": "1",
+                "RNA ID": "RNA_0043_H09",
+                "plate_barcode": "RNA_0043",
+                "source": "Alderley",
+                "coordinate": "H09",
+                "line_number": 2,
+                "Result": "Positive",
+                "file_name": "some file",
+                "file_name_date": None,
+                "created_at": now,
+                "updated_at": now,
+                "Lab ID": None,
+                "filtered_positive": True,
+                "filtered_positive_version": "v2.3",
+                "filtered_positive_timestamp": now,
```

Comment:
This makes the tests more brittle: you expect the two separate calls to `datetime.now()` to be the same, but in reality finite amount of time passes between the two calls, hence they could return different values

As below for other changes

Reply:
It's using the [freezegun](https://pypi.org/project/pytest-freezegun/) plugin though, to freeze time.

Reply:
Ah. That makes sense